src/components/solverComponents/SAT_Components/SAT_TextArea.tsx

```diff
@@ -0,0 +1,42 @@
+import 'prismjs/themes/prism-solarizedlight.css'
+import React from "react";
+import Editor from "react-simple-code-editor";
+import { highlight } from 'prismjs/components/prism-core';
+import { Container } from "@chakra-ui/react";
+import { SAT_language } from "./prism-SAT.js"
+
+interface ISAT_TextAreaProps {
+
+}
+
+interface ISAT_TextAreaState {
+    problemString: string;
+}
+
+export class SAT_TextArea extends React.Component<ISAT_TextAreaProps, ISAT_TextAreaState> {
+    constructor(props: ISAT_TextAreaProps) {
+        super(props);
+        this.state = { problemString: "" };
+    }
+
+    setProblemString(code: string): void {
+        this.setState({problemString: code});
+    }
+
+    render() {
+        return (
+        <Container border="1px" borderColor="#AAAAAA" borderRadius="10px">
+            <Editor
+                value={this.state.problemString}
+                onValueChange={code => this.setProblemString(code)}
+                highlight={code => highlight(code, SAT_language)}
+                padding={10}
+                style={{
+                    fontFamily: '"Fira code", "Fira Mono", monospace',
```

Comment:
Fira code is not available on many operating systems by default. To enforce a consistent style, we should either switch to a different font set or provide the font through through the ProvideQ frontend with the [`firacode` package](https://www.npmjs.com/package/firacode)

Reply:
how do you provide the font set via the package? what do I have to do?

Reply:
added to polish page